securebanking-openbanking-uk-rcs-server/src/test/java/com/forgerock/securebanking/openbanking/uk/rcs/testsupport/JwtHelper.java

```diff
@@ -0,0 +1,84 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.securebanking.openbanking.uk.rcs.testsupport;
+
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSObject;
+import com.nimbusds.jose.Payload;
+import com.nimbusds.jose.crypto.MACSigner;
+import lombok.SneakyThrows;
+import org.apache.commons.lang3.RandomStringUtils;
+
+/**
+ * Helps create JWTs with certain values populated dynamically.
+ */
+public class JwtHelper {
+
+    /**
+     * Creates a JWT that mimics the value that's sent from the UI when the user grants consent.
+     *
+     * @param amPath The path to the WireMock instance, simulating AM. This is used for the `consentApprovalRedirectUri`.
+     * @param serverPort The dynamic port that WireMock is running on.
+     * @param clientId The ID of the TPP client.
+     * @param intentId The Open Banking intent ID.
+     * @param username The PSU's username.
+     * @return A {@link String} representation of the consent decision JWT.
+     */
+    @SneakyThrows
+    public static String consentDecisionJwt(String amPath, int serverPort, String clientId, String intentId, String username) {
+        String amUri = "http://localhost:" + serverPort + amPath;
+        String payload = "{\n" +
+                "\"clientId\":\"" + clientId + "\"," +
+                "\"iss\":\"" + amUri + "\"," +
+                "\"csrf\":\"Tjgj3HTzRl2vn+TmIlPhizb9gqdfR7Zn2a/r5iH+syA=\"," +
+                "\"client_description\":\"\"," +
+                "\"aud\":\"forgerock-rcs\"," +
+                "\"save_consent_enabled\":true," +
+                "\"claims\":{" +
+                "\"id_token\":{" +
+                "\"acr\":{" +
+                "\"value\":\"urn:openbanking:psd2:sca\"," +
+                "\"essential\":true" +
+                "}," +
+                "\"openbanking_intent_id\":{" +
+                "\"value\":\"" + intentId + "\"," +
+                "\"essential\":true" +
+                "}" +
+                "}," +
+                "\"userinfo\":{" +
+                "\"openbanking_intent_id\":{" +
+                "\"value\":\"" + intentId + "\"," +
+                "\"essential\":true" +
+                "}" +
+                "}" +
+                "}," +
+                "\"scopes\":{" +
+                "\"accounts\":\"accounts\"," +
+                "\"openid\":\"openid\"," +
+                "\"payments\":\"payments\"" +
+                "}," +
+                "\"exp\":1614954054," +
+                "\"iat\":1614953874," +
+                "\"client_name\":\"bdddb430-3160-4aeb-85b9-ddebcb0b8ba2\"," +
+                "\"consentApprovalRedirectUri\":\"" + amUri + "?response_type=code%20id_token&client_id=" + clientId + "\"," +
```

Comment:
Some of the fiddly stuff I mentioned the other day - this has to point to the dynamically created WireMock instance. 😱 

I'm not a massive fan of this class with this brittle JSON and am tempted to improve it, but it's only used by tests.